streamer/src/sendmmsg.rs

```diff
@@ -278,12 +288,108 @@ mod tests {
     }
 
     #[test]
-    fn test_message_header_from_packet() {
-        let packets: Vec<_> = (0..2).map(|_| vec![0u8; PACKET_DATA_SIZE]).collect();
+    fn test_intermediate_failures_mismatched_bind() {
+        let packets: Vec<_> = (0..3).map(|_| vec![0u8; PACKET_DATA_SIZE]).collect();
         let ip4 = SocketAddr::new(IpAddr::V4(Ipv4Addr::LOCALHOST), 8080);
         let ip6 = SocketAddr::new(IpAddr::V6(Ipv6Addr::LOCALHOST), 8080);
-        let packet_refs: Vec<_> = vec![(&packets[0], &ip4), (&packets[1], &ip6)];
+        let packet_refs: Vec<_> = vec![
+            (&packets[0], &ip4),
+            (&packets[1], &ip6),
+            (&packets[2], &ip4),
+        ];
+        let dest_refs: Vec<_> = vec![&ip4, &ip6, &ip4];
+
         let sender = UdpSocket::bind(":::0").expect("bind");
-        batch_send(&sender, &packet_refs).unwrap();
+        if let Err((ioerror, num_sent)) = batch_send(&sender, &packet_refs) {
+            assert_matches!(ioerror.kind(), ErrorKind::InvalidInput);
```

Comment:
We can avoid the dependency and instead use `assert!(matches!(value, pattern))` in stable Rust

Reply:
done